### ISSUE TITLE
Set active milestone for sprint/iteration

### ISSUE THREAD

**Issue by agibson22** [waffleio/waffle.io#1709]
From customer:
A feature request, would be really cool to be able to set an active milestone. For example, I am using milestones like sprints (every 2 weeks). If I could set an active milestone/sprint then every ticket I move into ready/in progress would automagically get assigned to that milestone.

**Comment by brettlangdon** [comment]
am I allowed to +1 my own request? :)

**Comment by jasonmp85** [comment]
:+1:

I use them for versions, so having a "current milestone/next release" concept that gets applied to anything moved to Ready would be great.

Also nice, but possibly more annoying to implement:

  * if I move something out of Ready back to Backlog, remove the milestone
  * When the "current milestone" is changed, all things in Ready or higher (i.e. between Ready and Done) have their milestone updated

**Comment by aisapatino** [comment]
Agreed, I use milestones for sprints and it would be great to be able to set 'Ready' to use a milestone instead of a label.

**Comment by electrical** [comment]
+1 from me. I use Milestones for version planning of features / fixes.

**Comment by mazikwyry** [comment]
+1

**Comment by ppf-mcmasty** [comment]
+1 and some additional features  

- Allow closed milestones 'summary' data to live in the metrics / reporting (i.e. number of issues, story sizes, etc)  

- Allow a `Sprint Start Date` to be set.  To enable the calculation of a Sprint Duration.  That is the difference between `Sprint Start Date` and `Milestone Close Date`.  

- Allow for `Estimated Completion Date`.  To enable calculation of variance between estimated and actual close dates.  

- Reporting:  A table view of Sprints (Milestones), Start Date, Number of Issues, Sum of Story Points,  Actual Close, Sprint Duration, Estimated Close, Duration Estimate Variance     

This will allow a calibration of Story Points, as well as starting to observe a correlation of the Story Points to actual duration, which will help in future sprint planning.

**Comment by emartinez-usgs** [comment]
:+1: 

A related feature for my purpose would be for the "Done" column to only show cards for the current milestone. We often work on the same project in back-to-back sprints so the cards from the previous sprint would clutter the "Done" column.

**Comment by nlf** [comment]
+1

**Comment by MikroAcse** [comment]
+1
